log: Explicitly ignoring return of memcpy

ceolin opened the pull request:
According with MISRA-C the value returned by a non-void function has
to be used. As memcpy return is almost useless, we are explicitly
ignoring it.

MISRA-C rule 17.7

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
log: Explicitly ignoring return of memcpy

According with MISRA-C the value returned by a non-void function has
to be used. As memcpy return is almost useless, we are explicitly
ignoring it.

MISRA-C rule 17.7

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

himanshujha199640 commented:
@ceolin 
How about this ;-)
```
diff --git a/scripts/coccinelle/ignore_return.cocci b/scripts/coccinelle/ignore_return.cocci
index a6712b037b..59b6a8472c 100644
--- a/scripts/coccinelle/ignore_return.cocci
+++ b/scripts/coccinelle/ignore_return.cocci
@@ -14,6 +14,10 @@ virtual patch
 expression e1,e2,e3;
 @@
 
+(
 + (void)
 memset(e1,e2,e3);
-
+|
++ (void)
+memcpy(e1,e2,e3);
+)
```
Apply this and try yourself.
And then sending the bulk changes on the same like we did for `memset`?

ceolin commented:
> +( + (void) memset(e1,e2,e3); - +| ++ (void) +memcpy(e1,e2,e3); +)

good point. I'll add it.

Commit:
coccinelle: Updating ignore_return to support memcpy

The return of memcpy was being ignored just like memset's return. Just
adding it to coccinelle script.

MISRA-C rule 17.7

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
lib: posix: Explicitly ignoring return of memcpy

According with MISRA-C the value returned by a non-void function has
to be used. As memcpy return is almost useless, we are explicitly
ignoring it.

MISRA-C rule 17.7

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
arch: arm: Explicitly ignoring return of memcpy

According with MISRA-C the value returned by a non-void function has
to be used. As memcpy return is almost useless, we are explicitly
ignoring it.

MISRA-C rule 17.7

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>
